### ISSUE TITLE
Site won't build on Netlify. No errors, but a timeout.

### ISSUE THREAD

**Issue by danboyle8637** [gatsbyjs/gatsby#15275]
## Description

I'm trying to deploy to Netlify but my site won't build with their deploy process. It times out after generating image thumbnails. See below on Netlify Deploy logs.

Locally it builds fine with no errors.

I updated my packages after deleting a page I didn't need anymore and removing the components and queries for the page. That was the only change I made, but it seems updating my packages broke something.

I don't get any error message except you'll see in the Netlify deploy logs...

Browserslist: caniuse-lite is outdated. Please run next command `yarn upgrade caniuse-lite browserslist`

I've done some research on this but not sure if that is responsible for the build problem.

### Steps to reproduce

It's hard to reproduce unless you deploy to Netlify. But...

Here's a link to the folder Netlify uses for deploying:
https://github.com/danboyle8637/thistimefitness/tree/static

### Expected result

What should happen?

A successful build of the site.

### Actual result

Build times out.

### Netlify Deploy Logs

```
11:04:40 AM: Build ready to start
11:04:42 AM: build-image version: 8e315e54bc4032a32e73290be556cde4f8348c12
11:04:42 AM: build-image tag: v2.8.2
11:04:42 AM: buildbot version: b925d11411cabfe7b120501027bd0e96dbc28dde
11:04:42 AM: Fetching cached dependencies
11:04:43 AM: Starting to download cache of 248.0MB
11:04:44 AM: Finished downloading cache in 1.508065247s
11:04:44 AM: Starting to extract cache
11:04:53 AM: Finished extracting cache in 8.61190213s
11:04:53 AM: Finished fetching cache in 10.27178892s
11:04:53 AM: Starting to prepare the repo for build
11:04:53 AM: Preparing Git Reference refs/heads/static
11:04:54 AM: Starting build script
11:04:54 AM: Installing dependencies
11:04:55 AM: Started restoring cached node version
11:04:56 AM: Finished restoring cached node version
11:04:56 AM: v8.16.0 is already installed.
11:04:57 AM: Now using node v8.16.0 (npm v6.4.1)
11:04:58 AM: Attempting ruby version 2.3.6, read from environment
11:04:59 AM: Using ruby version 2.3.6
11:04:59 AM: Using PHP version 5.6
11:04:59 AM: Started restoring cached node modules
11:04:59 AM: Finished restoring cached node modules
11:04:59 AM: Started restoring cached yarn cache
11:04:59 AM: Finished restoring cached yarn cache
11:05:00 AM: Installing NPM modules using Yarn version 1.3.2
11:05:01 AM: yarn install v1.3.2
11:05:01 AM: [1/4] Resolving packages...
11:05:02 AM: warning gatsby > @hapi/joi@14.5.0: This version is no longer maintained
11:05:07 AM: [2/4] Fetching packages...
11:05:30 AM: info fsevents@1.2.4: The platform "linux" is incompatible with this module.
11:05:30 AM: info "fsevents@1.2.4" is an optional dependency and failed compatibility check. Excluding it from installation.
11:05:30 AM: info fsevents@1.2.9: The platform "linux" is incompatible with this module.
11:05:30 AM: info "fsevents@1.2.9" is an optional dependency and failed compatibility check. Excluding it from installation.
11:05:30 AM: [3/4] Linking dependencies...
11:05:30 AM: warning "gatsby > mini-css-extract-plugin > schema-utils > ajv-errors@1.0.1" has unmet peer dependency "ajv@>=5.0.0".
11:05:30 AM: warning "gatsby > pnp-webpack-plugin > ts-pnp@1.1.2" has unmet peer dependency "typescript@*".
11:05:30 AM: warning "gatsby-plugin-transition-link > babel-preset-gatsby-package > @babel/preset-env@7.2.3" has unmet peer dependency "@babel/core@^7.0.0-0".
11:05:30 AM: warning "gatsby-plugin-transition-link > babel-preset-gatsby-package > @babel/preset-env > @babel/plugin-proposal-object-rest-spread@7.2.0" has unmet peer dependency "@babel/core@^7.0.0-0".
11:05:30 AM: warning "gatsby-plugin-transition-link > babel-preset-gatsby-package > @babel/preset-env > @babel/plugin-proposal-unicode-property-regex@7.2.0" has unmet peer dependency "@babel/core@^7.0.0-0".
11:05:30 AM: warning "gatsby-plugin-transition-link > babel-preset-gatsby-package > @babel/preset-env > @babel/plugin-transform-async-to-generator@7.2.0" has unmet peer dependency "@babel/core@^7.0.0-0".
11:05:30 AM: warning "gatsby-plugin-transition-link > babel-preset-gatsby-package > @babel/preset-env > @babel/plugin-transform-block-scoping@7.2.0" has unmet peer dependency "@babel/core@^7.0.0-0".
11:05:30 AM: warning "gatsby-plugin-transition-link > babel-preset-gatsby-package > @babel/preset-env > @babel/plugin-transform-classes@7.2.2" has unmet peer dependency "@babel/core@^7.0.0-0".
11:05:30 AM: warning "gatsby-plugin-transition-link > babel-preset-gatsby-package > @babel/preset-env > @babel/plugin-transform-destructuring@7.2.0" has unmet peer dependency "@babel/core@^7.0.0-0".
11:05:30 AM: warning "gatsby-plugin-transition-link > babel-preset-gatsby-package > @babel/preset-env > @babel/plugin-transform-dotall-regex@7.2.0" has unmet peer dependency "@babel/core@^7.0.0-0".
11:05:30 AM: warning "gatsby-plugin-transition-link > babel-preset-gatsby-package > @babel/preset-env > @babel/plugin-transform-for-of@7.2.0" has unmet peer dependency "@babel/core@^7.0.0-0".
11:05:30 AM: warning "gatsby-plugin-transition-link > babel-preset-gatsby-package > @babel/preset-env > @babel/plugin-transform-function-name@7.2.0" has unmet peer dependency "@babel/core@^7.0.0-0".
11:05:30 AM: warning "gatsby-plugin-transition-link > babel-preset-gatsby-package > @babel/preset-env > @babel/plugin-transform-modules-commonjs@7.2.0" has unmet peer dependency "@babel/core@^7.0.0-0".
11:05:30 AM: warning "gatsby-plugin-transition-link > babel-preset-gatsby-package > @babel/preset-env > @babel/plugin-transform-modules-systemjs@7.2.0" has unmet peer dependency "@babel/core@^7.0.0-0".
11:05:30 AM: warning "gatsby-plugin-transition-link > babel-preset-gatsby-package > @babel/preset-env > @babel/plugin-transform-new-target@7.0.0" has unmet peer dependency "@babel/core@^7.0.0-0".
11:05:30 AM: warning "gatsby-plugin-transition-link > babel-preset-gatsby-package > @babel/preset-env > @babel/plugin-transform-parameters@7.2.0" has unmet peer dependency "@babel/core@^7.0.0-0".
11:05:30 AM: warning "gatsby-plugin-transition-link > babel-preset-gatsby-package > @babel/preset-env > @babel/plugin-transform-regenerator@7.0.0" has unmet peer dependency "@babel/core@^7.0.0-0".
11:05:30 AM: warning "gatsby-plugin-transition-link > babel-preset-gatsby-package > @babel/preset-env > @babel/plugin-transform-template-literals@7.2.0" has unmet peer dependency "@babel/core@^7.0.0-0".
11:05:30 AM: warning "gatsby-plugin-transition-link > babel-preset-gatsby-package > @babel/preset-env > @babel/plugin-transform-unicode-regex@7.2.0" has unmet peer dependency "@babel/core@^7.0.0-0".
11:05:30 AM: warning "gatsby-plugin-transition-link > babel-preset-gatsby-package > @babel/plugin-proposal-optional-chaining@7.2.0" has unmet peer dependency "@babel/core@^7.0.0-0".
11:05:30 AM: warning "gatsby-plugin-transition-link > babel-preset-gatsby-package > @babel/plugin-proposal-optional-chaining > @babel/plugin-syntax-optional-chaining@7.2.0" has unmet peer dependency "@babel/core@^7.0.0-0".
11:05:30 AM: warning "gatsby-plugin-transition-link > babel-preset-gatsby-package > @babel/preset-flow@7.0.0" has unmet peer dependency "@babel/core@^7.0.0-0".
11:05:30 AM: warning "gatsby-source-graphql > graphql-tools@3.1.1" has unmet peer dependency "graphql@^0.13.0".
11:05:30 AM: warning "gatsby-source-graphql > graphql-tools > apollo-link@1.2.6" has unmet peer dependency "graphql@^0.11.3 || ^0.12.3 || ^0.13.0 || ^14.0.0".
11:05:30 AM: warning "gatsby-source-graphql > apollo-link@1.2.1" has unmet peer dependency "graphql@^0.11.3 || ^0.12.3 || ^0.13.0".
11:05:30 AM: warning "gatsby-source-graphql > apollo-link-http@1.5.15" has unmet peer dependency "graphql@^0.11.0 || ^0.12.0 || ^0.13.0 || ^14.0.0".
11:05:30 AM: warning "gatsby-source-graphql > apollo-link-http > apollo-link@1.2.12" has unmet peer dependency "graphql@^0.11.3 || ^0.12.3 || ^0.13.0 || ^14.0.0".
11:05:30 AM: warning "gatsby-source-graphql > apollo-link-http > apollo-link-http-common@0.2.14" has unmet peer dependency "graphql@^0.11.0 || ^0.12.0 || ^0.13.0 || ^14.0.0".
11:05:30 AM: warning "gatsby-source-graphql > apollo-link-http > apollo-link > apollo-utilities@1.3.2" has unmet peer dependency "graphql@^0.11.0 || ^0.12.0 || ^0.13.0 || ^14.0.0".
11:05:41 AM: [4/4] Building fresh packages...
11:05:45 AM: success Saved lockfile.
11:05:45 AM: Done in 44.08s.
11:05:45 AM: NPM modules installed using Yarn
11:05:46 AM: Started restoring cached go cache
11:05:47 AM: Finished restoring cached go cache
11:05:47 AM: Installing Go version 1.10
11:05:47 AM: unset GOOS;
11:05:47 AM: unset GOARCH;
11:05:47 AM: export GOROOT='/opt/buildhome/.gimme_cache/versions/go1.10.linux.amd64';
11:05:47 AM: export PATH="/opt/buildhome/.gimme_cache/versions/go1.10.linux.amd64/bin:${PATH}";
11:05:47 AM: go version >&2;
11:05:47 AM: export GIMME_ENV='/opt/buildhome/.gimme_cache/env/go1.10.linux.amd64.env';
11:05:47 AM: go version go1.10 linux/amd64
11:05:47 AM: Installing missing commands
11:05:47 AM: Verify run directory
11:05:47 AM: Executing user command: gatsby build
11:05:50 AM: success open and validate gatsby-configs — 0.007
11:05:51 AM: success load plugins — 1.096
11:05:51 AM: success onPreInit — 0.132
11:05:51 AM: success delete html and css files from previous builds — 0.008
11:05:51 AM: success initialize cache — 0.010
11:05:51 AM: success copy gatsby files — 0.025
11:05:51 AM: success onPreBootstrap — 0.015
11:05:54 AM: success source and transform nodes — 2.182
11:05:54 AM: success building schema — 0.390
11:05:54 AM: success createPages — 0.444
11:05:55 AM: success createPagesStatefully — 0.085
11:05:55 AM: success onPreExtractQueries — 0.003
11:05:55 AM: success update schema — 0.035
11:05:55 AM: success extract queries from components — 0.361
11:05:55 AM: success write out requires — 0.005
11:05:55 AM: success write out redirect data — 0.001
11:05:55 AM: success Build manifest and related icons — 0.210
11:05:55 AM: success onPostBootstrap — 0.211
11:05:55 AM: info bootstrap finished - 7.897 s
11:05:55 AM: success Generating image thumbnails — 9/9 - 0.099 s
11:05:56 AM: success run static queries — 0.859 — 8/8 9.32 queries/second
11:17:42 AM: Browserslist: caniuse-lite is outdated. Please run next command `yarn upgrade caniuse-lite browserslist`
11:18:02 AM: success Building production JavaScript and CSS bundles — 726.509
11:18:03 AM: success Rewriting compilation hashes — 0.001
11:18:14 AM: success run page queries — 11.189 — 45/45 4.02 queries/second
11:18:22 AM: success Generating image thumbnails — 188/188 - 9.738 s
11:29:05 AM: success Building static HTML for pages — 643.712 — 45/45 30.94 pages/second
11:29:05 AM: info Done building in 1398.25 sec
11:29:06 AM: failed during stage 'building site': Command did not finish within the time limit
11:29:06 AM: Execution timed out after 15m0s
11:29:06 AM: Error running command: Command did not finish within the time limit
11:29:06 AM: Failing build: Failed to build site
11:29:06 AM: Finished processing build request in 24m23.995396395s
```

### Environment

```
System:
    OS: macOS 10.14.5
    CPU: (4) x64 Intel(R) Core(TM) i5-4570 CPU @ 3.20GHz
    Shell: 3.2.57 - /bin/bash
  Binaries:
    Node: 10.3.0 - /usr/local/bin/node
    Yarn: 1.16.0 - /usr/local/bin/yarn
    npm: 6.9.1 - /usr/local/bin/npm
  Languages:
    Python: 2.7.14 - /usr/local/bin/python
  Browsers:
    Chrome: 75.0.3770.100
    Firefox: 67.0.4
    Safari: 12.1.1
  npmPackages:
    gatsby: ^2.11.7 => 2.11.7
    gatsby-image: ^2.2.3 => 2.2.3
    gatsby-plugin-google-tagmanager: ^2.1.0 => 2.1.0
    gatsby-plugin-manifest: ^2.2.0 => 2.2.0
    gatsby-plugin-react-helmet: ^3.1.0 => 3.1.0
    gatsby-plugin-remove-serviceworker: ^1.0.0 => 1.0.0
    gatsby-plugin-sharp: ^2.2.1 => 2.2.1
    gatsby-plugin-styled-components: ^3.1.0 => 3.1.0
    gatsby-plugin-transition-link: ^1.12.4 => 1.12.4
    gatsby-plugin-typography: ^2.3.0 => 2.3.0
    gatsby-source-filesystem: ^2.1.1 => 2.1.1
    gatsby-source-graphql: ^2.1.0 => 2.1.0
    gatsby-transformer-remark: ^2.5.0 => 2.5.0
    gatsby-transformer-sharp: ^2.2.0 => 2.2.0
  npmGlobalPackages:
    gatsby-cli: 2.7.7
```

**Comment by wardpeet** [comment]
Does bumping your packages fix anything?

```
{
  "name": "gatsby-starter-default",
  "private": true,
  "description": "A simple starter to get up and developing quickly with Gatsby",
  "version": "0.1.0",
  "author": "Kyle Mathews <mathews.kyle@gmail.com>",
  "dependencies": {
    "babel-plugin-styled-components": "^1.10.0",
    "gatsby": "^2.12.0",
    "gatsby-image": "^2.2.3",
    "gatsby-plugin-manifest": "^2.2.0",
    "gatsby-plugin-offline": "^2.2.0",
    "gatsby-plugin-react-helmet": "^3.1.0",
    "gatsby-plugin-sharp": "^2.2.1",
    "gatsby-plugin-styled-components": "^3.0.5",
    "gatsby-plugin-typography": "^2.2.5",
    "gatsby-source-filesystem": "^2.1.1",
    "gatsby-transformer-remark": "^2.2.0",
    "gatsby-transformer-sharp": "^2.2.0",
    "gsap": "^2.0.2",
    "prop-types": "^15.7.2",
    "react": "^16.8.6",
    "react-dom": "^16.8.6",
    "react-helmet": "^5.2.1",
    "react-typography": "^0.16.18",
    "styled-components": "^4.1.3",
    "typography": "^0.16.18"
  },
  "keywords": [
    "gatsby"
  ],
  "license": "MIT",
  "scripts": {
    "build": "gatsby build",
    "develop": "gatsby develop",
    "start": "npm run develop",
    "format": "prettier --write \"src/**/*.js\"",
    "test": "echo \"Write tests! -> https://gatsby.app/unit-testing\""
  },
  "devDependencies": {
    "prettier": "^1.18.2"
  },
  "repository": {
    "type": "git",
    "url": "https://github.com/gatsbyjs/gatsby-starter-default"
  },
  "bugs": {
    "url": "https://github.com/gatsbyjs/gatsby/issues"
  }
}
```

**Comment by danboyle8637** [comment]
@wardpeet If you mean bumping by running npm update... yes I did that. 

This morning I did a deep update after learning npm update only does top level packages. 

I ran...

```
npm --depth 9999 update
```

This did make some updates. I redeployed the site to Netlify and it worked. 

I still get the error about:

```
Browserslist: caniuse-lite is outdated. Please run next command `yarn upgrade caniuse-lite browserslist`
```

But perhaps something with the update worked. No conclusive solution for me personally.
